### DISCUSSION TITLE
Razor Compiler Option for Target Method

### DISCUSSION THREAD

**Discussion by ShaunCurtis** [dotnet/aspnetcore#49135] [Q&A] [answered]
Is there any way to change the target method for the Razor compiler in Blazor?  So instead of writing compiled code to `BuildRenderTree`, it wrote it to say `BuildChildRenderTree` defined in the inherited class.

My reason for asking is such an option would make it much simpler to build out wrapper components using `ComponentBase`.  

Currently I write my own components implementing `IComponent`, so have solved the problem another way.  I'm OK with that.

However, there are regular questions on SO where the asker wants to build a wrapper.   Telling them to throw away `ComponentBase` and build their own library is normally beyond their skill level, or outside their control.

**Comment by javiercn** [comment, marked as answer]
No, there currently is no way to do this.
